packages/vite/src/node/build.ts

```diff
@@ -681,7 +681,8 @@ export function resolveLibFilename(
       'Name in package.json is required if option "build.lib.fileName" is not provided.'
     )
 
-  return `${name}.${format}.js`
+  const extension = format === 'es' ? 'mjs' : 'js'
```

Comment:
This only checks for `es` format now, but it could check if the format includes `es` to support #6555